types/styled-system/index.d.ts

```diff
@@ -38,9 +39,9 @@ export interface styleFn {
 export interface LowLevelStylefunctionArguments {
     prop: string;
     cssProperty?: string;
+    alias?: string;
     key?: string;
-    getter?: () => any;
-    transformValue?: (n: string | number) => any;
+    transformValue?: (n: string | number, scale: Array<string | number>) => any;
```

Comment:
Hey! I was about to start working on this as it's been bothering me, and found your PR. Thanks so much for taking care of this 🙇. I'd like to hear your thoughts `transformValue` specifically:

I'm using this utility to simplify APIs for our design system's end users, for example, instead of `justify-content`, our grid-rows have a `center` prop (`boolean | boolean[]`). Would it be better to make this entire interface Generic with smart defaults?

```ts
interface LowLevelStylefunctionArguments<N = string | number, Scale = <Array<string | number>> {
  prop: string;
  cssProperty?: string;
  alias?: string;
  tranformValue: (n: N, scale: Scale) => any;
}
```

Reply:
This would make the following typesafe:

```ts
// theme values:
const buttonSizes = {
  xs: '8px',
  sm: '16px',
};

const buttonSize = style<string, typeof buttonSizes>({
  prop: 'size',
  cssProperty: 'padding',
  transformValue: (n, scale) => scale[n]
});

const gridRowCenter = style<boolean>({
  prop: 'center',
  cssProperty: 'justify-content',
  transformValue: shouldCenter => shouldCenter ? 'center' : 'flex-start'
});
```

Reply:
cc @jxnblk, thoughts on this change? I think it more closely captures `styled-system`'s behavior...

Reply:
@StevenLang Glad to help out! It's been bothering me as well. 😄 I don't have any problem with the changes you're suggesting. I can update this PR as soon as I have a few extra minutes

Reply:
But this is a change in the library as well right? Or just the typings?

Reply:
No just the typings. Scale has been available as a 2nd arg since 4.0.

Reply:
@peduarte relevant changelog; https://github.com/styled-system/styled-system/blob/master/CHANGELOG.md#v400-2019-02-23

Reply:
Ahhh yes makes sense 👍

Reply:
Not sure if this matters for the change here but scales can include nested objects or arrays, not just numbers and strings